src/senaite/core/z3cform/widgets/datetimewidget.py

```diff
@@ -0,0 +1,262 @@
+        widget.addFieldClass(self)
+
+    def to_localized_time(self, time, long_format=None, time_only=None):
+        """Convert time to localized time
+        """
+        if dtime.is_dt(time):
+            # NOTE: ts.ulocalized_time converts the value into a DateTime
+            #       object, which always uses the current timezone without
+            #       daylight savings, which might result in an offset.
+            time = time.isoformat()
+        ts = api.get_tool("translation_service")
+        long_format = True if self.show_time else False
+        return ts.ulocalized_time(
+            time, long_format, time_only, self.context, domain="senaite.core")
+
+    def get_display_value(self):
+        """Returns the localized date value
+        """
+        value = self.value
+        dm = queryMultiAdapter((self.context, self.field), IDataManager)
+        if dm:
+            # extract the object from the database
+            value = dm.query()
+        return self.to_localized_time(value)
+
+    def to_datetime(self, value):
+        """convert date string to datetime object with tz
+
+        :param value: date or datetime
+        :type value: string or datetime object
+        :returns: datetime object
+        """
+        default = self.field.missing_value
+        timezone = self.default_timezone
+        return to_datetime(value, timezone=timezone, default=default)
+
+    def get_date(self, value):
+        """Return only the date part of the value
+
+        :returns: date string
+        """
+        dt = self.to_datetime(value)
+        if not dt:
+            return u""
+        return dt.strftime(DATE_FORMAT)
+
+    def get_time(self, value):
+        """Return only the time part of the value
+
+        :returns: time string
+        """
+        dt = self.to_datetime(value)
+        if not dt:
+            return u""
+        return dt.strftime(HOUR_FORMAT)
+
+    def date_now(self, offset=0):
+        """Get the current date without time component
+        """
+        ts = datetime.now().strftime(DATE_FORMAT)
+        dt = datetime.strptime(ts, DATE_FORMAT)
+        if offset:
+            dt = dt + timedelta(offset)
+        return dt
+
+    def get_max(self):
+        """Return the max allowed date in the future
+
+        :returns: date string
+        """
+        now = self.date_now()
+        return now.strftime(DATE_FORMAT)
+
+    def get_min(self):
+        """Return the min allowed date in the past
+
+        :returns: date string
+        """
+        now = self.date_now()
+        return now.strftime(DATE_FORMAT)
+
+    def attrs(self):
+        """Return the template attributes for the date field
+
+        :returns: dictionary of HTML attributes
+        """
+        attrs = {}
+        if self.datepicker_nofuture:
+            attrs["max"] = self.get_max()
+        if self.datepicker_nopast:
+            attrs["min"] = self.get_min()
+        return attrs
```

Comment:
These `get_min` and `get_max` are confusing, cause both functions have same implementation (return the string that represents current date without time component). I suggest to remove these two functions `get_min` and `get_max` and in `attrs` function, simply do:

```python
today = self.date_now()
today = today.strftime(DATE_FORMAT)
if self.datepicker_nofuture:
    attrs["max"] = today
if self.datepicker_nopast:
    attrs["min"] = today
return attrs
```

Reply:
Yes, you're right. I am wondering why I left it like this.
 As far as I remember I was struggling a bit that the max date was not selectable in the picker unless I added +1 day to it...